### DISCUSSION TITLE
How can I define in which directions velocities are generated by a thermostat?

### DISCUSSION THREAD

**Discussion by aldorvasq** [glotzerlab/hoomd-blue#1893] [Support] [answered]
Hello HOOMD users,

I am currently performing some simulations in the NVT ensemble that involve velocity generation via a thermostat (Langevin) in a specific direction only, say the y-direction, while leaving the other components (x and z) unconstrained. I have looked through the documentation and, to my best knowledge, I could not find anything that could do that. I thought of employing a filter of type  ``hoomd.filter.CustomFilter()`` to the thermostat but it is not obvious to me how that could be implemented for this scenario. Is there an easier way to implement the velocity generation in a specific direction?

Thank you,

Aldo V.

**Comment by joaander** [comment]
Langevin operates by applying forces to the particles. The guide ["How to apply arbitrary forces in MD"](https://hoomd-blue.readthedocs.io/en/v4.8.2/howto/custom-md-potential.html) describes the ways in which you can implement your custom force computation.

**Reply by aldorvasq** [reply]
Thank you @joaander for the prompt response,

From what I understood from your suggestion, should the proper implementation of what I am looking for then be to use ``hoomd.md.methods.ConstantVolume(thermostat=None)`` for the time integration and then remake the Langevin thermostat (excluding the forces by non-bonded and bonded potentials) from scratch as an arbitrary force where I redefine the random forces (in the specific direction I want) and the damping term?. I am not really looking to add additional forces to the system but to restrain which direction the forces (and the damping term) are applied by ``hoomd.md.methods.Langevin()``. Something similar to compute temp/partial in LAMMPS (https://docs.lammps.org/compute_temp_partial.html).

**Reply by joaander** [reply, marked as answer]
Your assessment of the HOOMD-blue documentation was correct. HOOMD-blue does not provide the option you describe. Therefore, the behavior your desire is custom. You can achieve your desired custom behavior with a custom force or by modifying the C++ class `TwoStepLangevin`.
